eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -53,6 +54,31 @@ jobs:
           packageType: sdk
           version: "$(DotNetCoreSDKVersion)"
 
+      # TODO: Remove when eng/New-TestResources.ps1 bootstraps the Az module.
+      - pwsh: Install-Module -Name Az -Scope CurrentUser -Force
+        displayName: "Install Azure PowerShell module"
+
+        # Chomp the string literal to more easily read and update the New-TestResources parameters.
+      - pwsh: >
+          $baseName = 't' + (New-Guid).ToString('n').Substring(0, 16);
```

Comment:
Why do we generate the base name here and not as part of the script?

Reply:
We require a `baseName` in the script so users will use scope their resources to something recognizable, like a username. One could probably make a reasonable argument that the `baseName` generated here is the default and devs could set it directly.

Reply:
Yeah, guid is not something recognizable. I think we can do a better job by default as part of the script.

Reply:
I don't want to generate it in the script because, as Daniel pointed out, the user needs to supply it. In fact, if anything, we could generate the name fully outside the script: whatever the user passes in is the resource name (still probably "rg-" as a prefix) and resources base name. The main point of this script is to be user-friendly. ID generation in the script with essential no input into the name is not user-friendly.